fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,157 @@
+            )
+        elif target_type == BINARY:
+            binary_transformer = Pipeline(
+                steps=[("imputer", SimpleImputer(strategy="most_frequent")), ("encoder", OrdinalEncoder())]
+            )
+            self.target_column_transformer = ColumnTransformer(
+                transformers=[("bin", binary_transformer, [target])], remainder="drop"
+            )
+        elif target_type == ORDINAL:
+            categorical_transformer = Pipeline(
+                steps=[
+                    (
+                        "encoder",
+                        OrdinalEncoder(
+                            unknown_value=len(target_categories) + 1,
+                            handle_unknown="use_encoded_value",
+                            categories=[target_categories],
+                        ),
+                    )
+                ]
+            )
+            self.target_column_transformer = ColumnTransformer(
+                transformers=[("cat", categorical_transformer, [target])], remainder="drop"
+            )
+
+    def initialize_default_data_transformer(self, tab_feature_encoder: TabularFeaturesInfoEncoder) -> None:
+        """
+        Initialize a default ColumnTransformer for the data columns
+        (i.e., all columns except the target column)
+        """
+        self.categories = tab_feature_encoder.get_categories_list()
+        self.target_column = tab_feature_encoder.get_target()
+        self.type_to_features: Dict[str, List[str]] = tab_feature_encoder.type_to_features()
+        self.default_fill_values: Dict[str, Scalar] = tab_feature_encoder.get_all_default_fill_values()
+        self.vocabulary = tab_feature_encoder.get_vocabulary()
+
+        self.transformers[NUMERIC] = Pipeline(
+            steps=[("imputer", SimpleImputer(strategy="mean")), ("scaler", MinMaxScaler())]
+        )
+        self.transformers[BINARY] = Pipeline(
+            steps=[("imputer", SimpleImputer(strategy="most_frequent")), ("encoder", OrdinalEncoder())]
+        )
+        self.transformers[ORDINAL] = Pipeline(
+            steps=[("encoder", OneHotEncoder(handle_unknown="ignore", categories=self.categories))]
+        )
+        self.transformers[STRING] = Pipeline(
+            steps=[("vectorizer", StringColumnTransformer(TfidfVectorizer(vocabulary=self.vocabulary)))]
+        )
+
+        self.data_column_transformer = ColumnTransformer(
+            transformers=[
+                ("num", self.transformers[NUMERIC], self.type_to_features[NUMERIC]),
+                ("bin", self.transformers[BINARY], self.type_to_features[BINARY]),
+                ("str_vectorizer", self.transformers[STRING], self.type_to_features[STRING]),
+                ("cat", self.transformers[ORDINAL], self.type_to_features[ORDINAL]),
+            ],
+            remainder="drop",
+        )
+
+    def set_data_transformer(self, data_type: str, transformer: Pipeline) -> None:
```

Comment:
Maybe we can extend this function to add transformations for specific column names beyond the data_type restriction. So we would be able to apply any customized pipeline on any of the feature columns.